core/src/main/scala/io/projectglow/transformers/pipe/Piper.scala

```diff
@@ -70,7 +72,7 @@ private[projectglow] object Piper extends GlowLogging {
       .mapPartitions { it =>
         new PipeIterator(cmd, env, it, informatter, outputformatter)
       }
-      .cache()
+      .persist(StorageLevel.DISK_ONLY)
```

Comment:
You might want to try `MEMORY_AND_DISK`. I believe it works like `DISK_ONLY` but keeps blocks in memory if they fit.